src/Reflection/ReflectionClass.php

```diff
@@ -279,6 +285,55 @@ static function (ReflectionClass $ancestor) : array {
         );
     }
 
+    /**
+     * @return ReflectionMethod[]|null
+     */
+    private function createMethodsFromTrait(ReflectionMethod $method) : ?array
+    {
+        $traitAliases     = $this->getTraitAliases();
+        $traitPrecedences = $this->getTraitPrecedences();
+
+        $methodAst = $method->getAst();
+        assert($methodAst instanceof ClassMethod);
+
+        $methodHash = $this->methodHash($method->getDeclaringClass()->getName(), $method->getName());
+
+        $aliases = [];
+        foreach ($traitAliases as $aliasMethodName => $traitAliasDefinition) {
+            if ($methodHash !== $traitAliasDefinition) {
+                continue;
+            }
+
+            $aliases[] = ReflectionMethod::createFromNode(
+                $this->reflector,
+                $methodAst,
+                $method->getDeclaringClass()->getDeclaringNamespaceAst(),
+                $method->getDeclaringClass(),
+                $this,
+                $aliasMethodName
+            );
+        }
+
+        if ($aliases !== []) {
+            return $aliases;
+        }
+
+        if (array_key_exists($methodHash, $traitPrecedences)) {
+            return null;
```

Comment:
Would it make sense to return `[]` here too? Or is there a specific rationale for returning `null|[]`?

Reply:
There were approximately 50 versions of the code before I sent the PR - I've almost killed @ondrejmirtes for finding more and more strange bugs :)

So it was just a residue.

Removed and squashed.

Reply:
Don't blame me, blame PHP :)

Reply:
Blame decades of development in a dynlang :P

Reply:
We also reported one PHP bug https://bugs.php.net/bug.php?id=79605 (already fixed in PHP 8)